src/librustdoc/html/format.rs

```diff
@@ -498,13 +488,9 @@ fn resolved_path(w: &mut fmt::Formatter, did: DefId, path: &clean::Path,
                     } else {
                         root.push_str(&seg.name);
                         root.push_str("/");
-                        if is_not_debug {
```

Comment:
With `w.alternate() == true` it does `write!(w, "{}::", seg.name)?;`, as can be seen above. The `is_not_debug == false` case did exactly the same thing.